### PR TITLE
qhT accessors API

### LINKED ISSUE #86: constructor/accessor API to be friendlier to non-C languages?
It would be nice to have an API for libqhull_r that is easier to call from non-C languages that can call out to compiled C libraries (e.g. Python's `ctypes` or Julia's `ccall`), but which can't parse C header files.  This seems to require two things:

* add `create/destroy` methods for allocating a pointer to a `struct qhT`, which is a [huge datastructure](https://github.com/qhull/qhull/blob/d0e8064088e0d51f7df9b380b22d66a3c6ba0c9a/src/libqhull_r/libqhull_r.h#L468-L861) whose size is hard to predict without running a C compiler, particularly as it has numerous [`jmp_buf` fields](https://github.com/qhull/qhull/blob/d0e8064088e0d51f7df9b380b22d66a3c6ba0c9a/src/libqhull_r/libqhull_r.h#L672) (whose size is OS-dependent and perhaps compiler-dependent).

* add accessor functions that take a `qhT*` and allow you to get fields that are part of the documented API / externally useful (as opposed to stuff that is only used internally by qhull).   You could maybe have a callback API for things like `FORALLpoints`, but those data structures are not as terrible as `qhT*` to access externally.

For example, Julia has a relatively simple way to add [dependencies on compiled binary packages](https://julialang.org/blog/2019/11/artifacts/), and in fact already has precompiled binaries of [Qhull](https://github.com/JuliaPackaging/Yggdrasil/tree/master/Q/Qhull) built for [all supported platforms](https://github.com/JuliaPackaging/Yggdrasil).  This would make it easy for Julia packages to depend on Qhull with automated installation, even on user systems that don't have a C compiler, except that it's currently not possible to call Qhull without having a C compiler handy to parse the header file.   (e.g. pyhull needs a C compiler.)  One way around this is to build our own C wrapper library and distribute that (gridap/MiniQhull.jl#5), but it seems much more natural to include such an API into Qhull itself.

Is this something you would be interested in?   If so, I and others could work on a PR.

cc @fverdugo, @blegat

### PR DESCRIPTION
**Pull request by stevengj**
Minimally fixes #86.

The main obstacle to accessing libqhull_r without a C compiler is the `struct qhT` type, which is practically impossible to define without a C compiler because of the `jmp_buf` fields.   So this PR adds `qh_alloc_qh` and `qh_free_qh` functions to `malloc` and `free` a pointer to `qhT`, as well as a minimal set of accessor functions — enough to emulate the various `FOR*` macros.

(My assumption is that the `facetT` and `pointT` structs, which are only slightly unwieldy, can be declared without too much trouble in a non-C language like Julia, so they don't need their own accessor API.)

cc @fverdugo, @blegat — are any more accessors needed?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #   See README.txt
 #   For qhulltest of the C++ interface, use Qt project file at src/qhull-all.pro
 #   For static builds, a simple alternative is src/libqhull_r/Makefile
-#   
+#
```
Comment (stevengj) [inline Makefile:7]: Sorry for the whitespace changes — my editor deletes trailing whitespace by default.  Note that it is possible to [view the diff ignoring whitespace](https://stackoverflow.com/questions/37007300/how-to-ignore-whitespace-in-github-when-comparing) in various ways on github, e.g.  by appending `?w=1` to the URL.

**Comment by blegat** [comment]
Thanks! Qhull.jl would also need accessors for `totarea` and `totvol`. I also think that getters and setters for `hasAreaVolume` and `hasTriangulation` would be useful because in case you make some change to the polytope, you should be able to reset these to `false`. I don't know if it's possible to make some change to the polytope but it seems [scipy.spatial sets this bool so we might need to do it as well](https://github.com/scipy/scipy/blob/edb50b09e92cc1b493242076b3f63e89397032e4/scipy/spatial/qhull.pyx#L508)

**Comment by stevengj** [comment]
@blegat, thanks, I've added all of those suggestions.

**Comment by cbbarber** [comment]
Steven and Benolt,

Many thanks for your good addition to the Qhull library.

Please resubmit.  There are several problems

1) Rename accessors.c to accessors_r.c -- This follows the convention used by the other files in libqhull_r.   The next release of Qhull will add another file, libqhull/accessors.c, with the same content but using a statically allocated qhT.

2) Alphabetize the functions in accessors_r.c and the corresponding prototypes in libqhull_r.h 

3) Keep the deletion of trailing spaces.   For the next release, I'll consider deleting trailing spaces everywhere.

4) As done in user_eg_r.c, add the following to qh_alloc_qh.  It helps catch errors.   The next release will add qh_zero to user_eg2_r.c

  QHULL_LIB_CHECK
  qh_zero(qh, errfile);

5) Add a comment to qh_free_qh that qh_freeqhull and qh_memfreeshort should be called before calling qh_free_qh.   See user_eg.c for examples.  These routines and error checks perform a clean release of qhull's memory.

**Comment by stevengj** [comment]
@cbbarber, thanks for your comments.  I have made the changes you requested.

**Comment by stevengj** [comment]
@cbbarber, do you have any further comments?

**Comment by cbbarber** [comment]
Looks good.  I've squashed and merged this pull request.

One question:  Elsewhere in Qhull, malloc and free are called via qh_free and qh_malloc in qh-usermem_r.c.   This allows a custom malloc/free to be substituted for the system malloc/free.  Should qh_alloc_qh and qh_free_qh do the same?

**Comment by stevengj** [comment]
Sure, I'll update it to use `qh_malloc/free` ASAP.

**Comment by stevengj** [comment]
Updated for `qh_malloc`.

**Pull request merged**